jdisc_http_service/src/main/java/com/yahoo/jdisc/http/filter/JdiscFilterRequest.java

```diff
@@ -113,6 +116,11 @@ public void setUserPrincipal(Principal principal) {
         this.parent.setUserPrincipal(principal);
     }
 
+    @Override
+    public Optional<X509Certificate[]> getClientCertificateChain() {
```

Comment:
Maybe this could return `List<X509Certificate>`? Easier to use and the caller doesn't have to wonder if there can be two empty cases: Empty optional and optional of an empty array.

Reply:
I agree with the arguments. I have changed the signature in the next commit.